src/App.jsx

```diff
@@ -1,58 +1,98 @@
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
-
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const initialGoods = [
+    'Dumplings',
+    'Carrot',
+    'Eggs',
+    'Ice cream',
+    'Apple',
+    'Bread',
+    'Fish',
+    'Honey',
+    'Jam',
+    'Garlic',
+  ];
+
+  const [goods, setGoods] = useState(initialGoods);
+  const [sortOrder, setSortOrder] = useState('initial');
+
+  const handleSort = (order) => {
+    let sortedGoods = [...goods];
+
+    switch (order) {
+      case 'alphabetical':
+        sortedGoods.sort();
+        break;
+      case 'length':
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      case 'reverse':
+        sortedGoods.reverse();
+        break;
+      case 'initial':
+        sortedGoods = initialGoods;
+        break;
+      default:
+        break;
+    }
+
+    setGoods(sortedGoods);
+    setSortOrder(order);
+  };
+
+  const showResetButton = sortOrder !== 'initial';
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortOrder === 'alphabetical' ? '' : 'is-light'}`}
+          onClick={() => handleSort('alphabetical')}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${sortOrder === 'length' ? '' : 'is-light'}`}
+          onClick={() => handleSort('length')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${sortOrder === 'reverse' ? '' : 'is-light'}`}
+          onClick={() => {
+            handleSort(sortOrder === 'reverse' ? 'initial' : 'reverse');
+          }
```

Comment:
Add the state for this button. We should have possibility to click sorting button and reverse button shouldn't change if we pressed it before